apps/aecontract/src/aect_contracts.erl

```diff
@@ -137,70 +137,30 @@ is_legal_serialization_at_height(?ABI_FATE_SOPHIA_1, Vsn, Height) ->
     aect_sophia:is_legal_serialization_at_height(Vsn, Height).
 
 is_legal_version_in_protocol(create, #{vm := ?VM_AEVM_SOPHIA_1, abi := ?ABI_AEVM_SOPHIA_1}, ProtocolVersion) ->
-    case ProtocolVersion of
-        ?ROMA_PROTOCOL_VSN -> true;
-        P when P >= ?MINERVA_PROTOCOL_VSN -> false
-    end;
+    version_valid_at(ProtocolVersion, ?ROMA_PROTOCOL_VSN);
 is_legal_version_in_protocol(create, #{vm := ?VM_AEVM_SOPHIA_2, abi := ?ABI_AEVM_SOPHIA_1}, ProtocolVersion) ->
-    case ProtocolVersion of
-        ?ROMA_PROTOCOL_VSN    -> false;
-        ?MINERVA_PROTOCOL_VSN -> true;
-        ?FORTUNA_PROTOCOL_VSN -> true;
-        P when P >= ?LIMA_PROTOCOL_VSN -> false
-    end;
+    version_valid_between(ProtocolVersion, ?MINERVA_PROTOCOL_VSN, ?FORTUNA_PROTOCOL_VSN);
 is_legal_version_in_protocol(create, #{vm := ?VM_AEVM_SOPHIA_3, abi := ?ABI_AEVM_SOPHIA_1}, ProtocolVersion) ->
-    case ProtocolVersion of
-        ?ROMA_PROTOCOL_VSN    -> false;
-        ?MINERVA_PROTOCOL_VSN -> false;
-        ?FORTUNA_PROTOCOL_VSN -> true;
-        P when P >= ?LIMA_PROTOCOL_VSN -> false
-    end;
+    version_valid_at(ProtocolVersion, ?FORTUNA_PROTOCOL_VSN);
 is_legal_version_in_protocol(create, #{vm := ?VM_AEVM_SOPHIA_4, abi := ?ABI_AEVM_SOPHIA_1}, ProtocolVersion) ->
-    case ProtocolVersion of
-        ?ROMA_PROTOCOL_VSN    -> false;
-        ?MINERVA_PROTOCOL_VSN -> false;
-        ?FORTUNA_PROTOCOL_VSN -> false;
-        ?LIMA_PROTOCOL_VSN    -> true;
-        P when P > ?LIMA_PROTOCOL_VSN -> true %% TODO: If you bump to VM_AEVM_SOPHIA_5 please turn VM_AEVM_SOPHIA_4 off for the new consensus protocol!
-    end;
+    version_valid_after(ProtocolVersion, ?LIMA_PROTOCOL_VSN); %% TODO: If you bump to VM_AEVM_SOPHIA_5 please turn VM_AEVM_SOPHIA_4 off for the new consensus protocol!
+    %%version_valid_between(ProtocolVersion, ?LIMA_PROTOCOL_VSN, ?FORK_INTRODUCING_VM_AEVM_SOPHIA_5);
 is_legal_version_in_protocol(create, #{vm := ?VM_FATE_SOPHIA_1, abi := ?ABI_FATE_SOPHIA_1}, ProtocolVersion) ->
-    case ProtocolVersion of
-        ?ROMA_PROTOCOL_VSN    -> false;
-        ?MINERVA_PROTOCOL_VSN -> false;
-        ?FORTUNA_PROTOCOL_VSN -> false;
-        ?LIMA_PROTOCOL_VSN    -> ?VM_FATE_SOPHIA_1_enabled; %% TODO: Revise this before release
-        P when P > ?LIMA_PROTOCOL_VSN -> ?VM_FATE_SOPHIA_1_enabled %% TODO: Revise this before release
-    end;
+    ?VM_FATE_SOPHIA_1_enabled and version_valid_after(ProtocolVersion, ?LIMA_PROTOCOL_VSN);
+is_legal_version_in_protocol(call, #{vm := ?VM_AEVM_SOLIDITY_1}, _ProtocolVersion) ->
+    ?VM_AEVM_SOLIDITY_1_enabled;
+is_legal_version_in_protocol(call, #{vm := ?VM_FATE_SOPHIA_1}, ProtocolVersion) ->
+    ?VM_FATE_SOPHIA_1_enabled and version_valid_at(ProtocolVersion, ?LIMA_PROTOCOL_VSN);
```

Comment:
If this is meant to be equivalent to before, this does not seem equivalent to me. Maybe:
```suggestion
    ?VM_FATE_SOPHIA_1_enabled and version_valid_after(ProtocolVersion, ?LIMA_PROTOCOL_VSN);
```
?